tests/SpecTests/Operation.php

```diff
@@ -723,7 +723,7 @@ private function getResultAssertionTypeForCollection()
                     return ResultExpectation::ASSERT_NOTHING;
                 }
 
-                return ResultExpectation::ASSERT_SAME_DOCUMENTS;
+                return ResultExpectation::ASSERT_DOCUMENTS_MATCH;
```

Comment:
If I understand correctly, you're making the assertion less strict, what's the reason?

Reply:
This is explained in the commit message for https://github.com/mongodb/mongo-php-library/pull/1064/commits/42bce4ea0e4c42b30ee7b8017ec76c537adae6d2. To quote:

> Allow extra fields in aggregate results
>
> Most operations that use ASSERT_SAME_DOCUMENT and ASSERT_SAME_DOCUMENTS should technically use ASSERT_MATCHES_DOCUMENT and ASSERT_DOCUMENTS_MATCH, respectively; however, this is the minimum change needed to satisfy new CSFLE spec tests. Related to dc1bb17.

[`assertSameDocuments()`](https://github.com/mongodb/mongo-php-library/blob/1.15.0/tests/TestCase.php#L95) does not permit extra fields in the actual value, while [`assertDocumentsMatch()`](https://github.com/mongodb/mongo-php-library/blob/1.15.0/tests/SpecTests/FunctionalTestCase.php#L91) does (see: [DocumentsMatchConstraint](https://github.com/mongodb/mongo-php-library/blob/1.15.0/tests/SpecTests/DocumentsMatchConstraint.php)).

The CSFLE legacy test runner inherits rules from the Transactions legacy test runner, which in turn inherits document matching rules from the [CRUD v1 runner](https://github.com/mongodb/specifications/blob/master/source/crud/tests/README.rst#evaluating-matches) -- which is where extra fields are permitted, among other things.

Per the commit message above, _more_ operations should be using the same logic here but I wanted to be conservative and avoid making extra changes to the legacy test runner (looking forward to the day we can just delete this code after porting everything to the unified format). No objections if you'd like to create a ticket to look into this at some point.